tests/rgw/sanity_rgw.py

```diff
@@ -24,40 +24,56 @@ def run(ceph_cluster, **kw):
     Args:
         ceph_cluster (ceph.ceph.Ceph): ceph cluster
     """
-    log.info("Running test")
     config = kw.get("config")
-    log.info("Running rgw tests %s" % config.get("test-version", "v2"))
+    log.info("Running RGW test version: %s", config.get("test-version", "v2"))
+
     rgw_ceph_object = ceph_cluster.get_ceph_object("rgw")
     run_io_verify = config.get("run_io_verify", False)
-    git_url = "https://github.com/red-hat-storage/ceph-qe-scripts.git"
-    branch = " -b master"
-    git_clone = "sudo git clone " + git_url + branch
     rgw_node = rgw_ceph_object.node
-    # cleanup any existing stale test dir
-    log.info("flushing iptables")
+
+    log.info("Flushing iptables")
     rgw_node.exec_command(cmd="sudo iptables -F", check_ec=False)
+
     test_folder = "rgw-tests"
-    test_folder_path = "~/{test_folder}".format(test_folder=test_folder)
-    rgw_node.exec_command(cmd="sudo yum install python3 -y", check_ec=False)
-    rgw_node.exec_command(cmd="sudo rm -rf " + test_folder)
-    rgw_node.exec_command(cmd="sudo mkdir " + test_folder)
-    rgw_node.exec_command(cmd="cd " + test_folder + " ; " + git_clone)
-    if ceph_cluster.containerized:
-        rgw_node.exec_command(cmd="sudo yum install ceph-radosgw -y")
-    rgw_node.exec_command(
-        cmd="sudo pip3 install -r {test_folder}/ceph-qe-scripts/rgw/requirements.txt".format(
-            test_folder=test_folder
+    test_folder_path = f"~/{test_folder}"
+    git_url = "https://github.com/red-hat-storage/ceph-qe-scripts.git"
+    git_clone = f"git clone {git_url} -b master"
+
+    # Clone the repository once for the entire test suite
+    pip_cmd = "venv/bin/pip"
```

Comment:
`venv` here is the virtual env name. and if we are separating out the cmd here I think it would be best to define another variable for virtualenv and use the same where ever possible. i,.e while creating venv like in line  51